Azure/sonic-buildimage
# DellEMC: Z9332f change SFP detection logic

**- Why I did it**

- Dynamically change EEPROM driver based on media type.
- Otherwise, EEPROM INFO and DOM INFO might not be fetched properly and will result in erroneous output.

**- How I did it**
Changed the optoe driver by reading optic EEPROM byte 
**- How to verify it**

- Check driver type in optic driver path(/sys/class/i2c-adapter/<i2c-path>/name.
- Take hexdump of optic EEPROM after OIR.
- Check sfpshow eeprom -d

**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**

- Change the SFP driver based on optic type. 

UT: 
[z9332f_sfp_detection_logic_UT.txt](https://github.com/Azure/sonic-buildimage/files/5723790/z9332f_sfp_detection_logic_UT.txt)




**- A picture of a cute animal (not mandatory but encouraged)**
